mmligneralign.py

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/python
+
+# mmligneralign.py
+
+
+import disk
+import os
+from collections import namedtuple
+
+import yasara
+
+# Represents a yasara object being aligned
+YObject = namedtuple("YObject", ["filename", "num", "mols"])
+
+
+def molecule(obj):
+  """Convert a YObject into an argument for MMLigner"""
+  chainIDs = obj.mols.replace("_", "")
+
+  if chainIDs == "":
+    return obj.filename
+
+  return obj.filename + ":" + chainIDs
+
+def execute_mmligner(mmligner_bin, objects):
+  """Execute MMLigner on provided inputs"""
+  import subprocess
+  mmlignercommand = [mmligner_bin,
+                     molecule(objects[0]),
+                     molecule(objects[1]),
+                     "--superpose"]
+
+  return subprocess.call(mmlignercommand)
```

Comment:
The issue is that the Windows command line does not seem to run like this.
C:\Users\albot\Documents\MobaXterm\home\mmligner64.exe C:\yasara\plg\mmligner_cache\Object1.pdb:A C:\yasara\plg\mmligner_cache\Object2.pdb:A --superpose
in the former version I worked around with this ugly string concatenation like "./"+'Object1.pdb'+':A'
maybe we find a way that the full path is also started when tweaking with the blackslashes or so.

Reply:
That's a bit surprising. I wonder why that's failing. We can sit together tomorrow to sort it out

Reply:
Alright, just pushed a commit using basenames for mmligner arguments

Reply:
indeed for tomorrow
C:\yasara\plg\mmligner_cache>C:\Users\albot\Documents\MobaXterm\home\mmligner64.exe C:\yasara\plg\mmligner_cache\Object1.pdb:A C:\yasara\plg\mmligner_cache\Object2.pdb:A --superpose
[0;37;1;44mMMLigner (v1.0.2): Minimum Message Length (MML) alignment of protein structures [0;0m
[0;1;37;1;46m   Authors:[0;1;37;1;46m    Collier, Allison, Lesk, Stuckey, Garcia de la Banda, Konagurthu. [0m
[1;37;1;46m   Reference:[0;1;37;1;46m  Bioinformatics. 33(7):1005-1013, 2017.                           [0m

::CmdLineError:: File 'C' does not exist
::CmdLineError:: File 'C' does not exist
    Usage: mmligner <pdb1>(:<chainIDs>) <pdb2>(:<chainIDs)
                      (--ivalue <fasta>)
                      ( -o  <output_prefix_string>)
                      ( --superpose)

We will need to google to fix this for the Windows paths.

I will go back to Acinetobase now. When reusing Thomas' CAMEL code, I made some progress :-)

Reply:
Oh, you are so quick, I did not even see the update with the basenames. And surprise 
![image](https://user-images.githubusercontent.com/38151284/75379829-51fcf780-58d6-11ea-9a0c-1be6dde3efd1.png)

Reply:
Ah great!